align/align_test.py

```diff
@@ -214,4 +231,312 @@ def test_multiply():
                 expand("torch_out_grad.pt"),
             ),
         ]
-    )
\ No newline at end of file
+    )
+
+# comment because of the probability of dropout
+# def test_dropout():
+#     out_dir = os.path.join(BASE_DIR, "dropout", "out")
+#     expand = prepend_dirname_fn(out_dir)
+#     align_tensors(
+#         [
+#             TensorAlignmentData(
+#                 "dropout_out",
+#                 expand("ff_out.pt"),
+#                 expand("torch_out.pt"),
+#             ),
+#             TensorAlignmentData(
+#                 "dropout_out_grad",
+#                 expand("ff_out_grad.pt"),
+#                 expand("torch_out_grad.pt"),
+#             ),
+#         ]
+#     )
+    
+def test_pool2d():
+    out_dir = os.path.join(BASE_DIR, "pool2d", "out")
+    expand = prepend_dirname_fn(out_dir)
+    align_tensors(
+        [
+            TensorAlignmentData(
+                "pool2d_out",
+                expand("ff_out.pt"),
+                expand("torch_out.pt"),
+            ),
+            TensorAlignmentData(
+                "pool2d_out_grad",
+                expand("ff_out_grad.pt"),
+                expand("torch_out_grad.pt"),
+            ),
+        ]
+    )
+    
+def test_reducesum():
+    out_dir = os.path.join(BASE_DIR, "reducesum", "out")
+    expand = prepend_dirname_fn(out_dir)
+    align_tensors(
+        [
+            TensorAlignmentData(
+                "reducesum_out",
+                expand("ff_out.pt"),
+                expand("torch_out.pt"),
+            ),
+            TensorAlignmentData(
+                "reducesum_out_grad",
+                expand("ff_out_grad.pt"),
+                expand("torch_out_grad.pt"),
+            ),
+        ]
+    )
+    
+def test_reshape():
+    out_dir = os.path.join(BASE_DIR, "reshape", "out")
+    expand = prepend_dirname_fn(out_dir)
+    align_tensors(
+        [
+            TensorAlignmentData(
+                "reshape_out",
+                expand("ff_out.pt"),
+                expand("torch_out.pt"),
+            ),
+            TensorAlignmentData(
+                "reshape_out_grad",
+                expand("ff_out_grad.pt"),
+                expand("torch_out_grad.pt"),
+            ),
+        ]
+    )
+    
+def test_flat():
+    out_dir = os.path.join(BASE_DIR, "flat", "out")
+    expand = prepend_dirname_fn(out_dir)
+    align_tensors(
+        [
+            TensorAlignmentData(
+                "flat_out",
+                expand("ff_out.pt"),
+                expand("torch_out.pt"),
+            ),
+            TensorAlignmentData(
+                "flat_out_grad",
+                expand("ff_out_grad.pt"),
+                expand("torch_out_grad.pt"),
+            ),
+        ]
+    )
+    
+def test_sin():
```

Comment:
Is there any way to factor out all the redundancy between these? I'm not sure there's a way to do it that doesn't add more complexity than it's worth, just wondering

Reply:
This initial readme contains a single operator test, do we need to test them separately?

Reply:
We should be able to test them separately if needed

Reply:
give me some time to think about it

Reply:
addressed